### ISSUE TITLE
Did not meet early stopping

### ISSUE THREAD

**Issue by Solly7** [microsoft/LightGBM#2840]
I am using python api 'train' to train my model in lightgbm but it always say 'Did not meet early stopping'. see below code

lgbm_params = {
    'objective': 'binary',
    'metric': 'auc','binary'
    'boosting': 'gbdt',
    'max_depth':3,
    'num_leaves': 7, 
    'learning_rate': 0.2,
    'min_data_in_leaf':20,
    'min_child_samples': 20,
    'verbose': 0,
    'first_metric_only': True 
}

lgtrain = lgb.Dataset(train, label=label)
lgvalid = lgb.Dataset(test.drop(['Is_Fraud'],axis=1), label=test['Is_Fraud'],reference=lgtrain)

lgb_clf = lgb.train(lgbm_params, lgtrain, num_boost_round=100 ,early_stopping_rounds=10,valid_sets=[lgtrain,lgvalid])

**Comment by StrikerRUS** [comment]
@Solly7 
I don't see any issue here. The fact that your training process doesn't meet early stopping maybe related to your data or/and params.

Please take a look at the following example, which confirms that early stopping works well with your params.

```
from sklearn.datasets import load_digits
from sklearn.model_selection import train_test_split

import lightgbm as lgb

X, y = load_digits(2, True)
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

lgbm_params = {
'objective': 'binary',
'metric': ['auc','binary'],
'boosting': 'gbdt',
'max_depth':3,
'num_leaves': 7,
'learning_rate': 0.2,
'min_data_in_leaf':20,
'min_child_samples': 20,
'verbose': 0,
'first_metric_only': True
}

lgtrain = lgb.Dataset(X_train, label=y_train)
lgvalid = lgb.Dataset(X_test, label=y_test, reference=lgtrain)

lgb_clf = lgb.train(lgbm_params, lgtrain, num_boost_round=100 ,early_stopping_rounds=10,valid_sets=[lgtrain,lgvalid])
```

```
[1]	training's auc: 0.998553	training's binary_logloss: 0.51952	valid_1's auc: 0.998456	valid_1's binary_logloss: 0.519994
Training until validation scores don't improve for 10 rounds
[2]	training's auc: 1	training's binary_logloss: 0.401461	valid_1's auc: 0.996911	valid_1's binary_logloss: 0.40376
[3]	training's auc: 1	training's binary_logloss: 0.316467	valid_1's auc: 1	valid_1's binary_logloss: 0.316695
[4]	training's auc: 1	training's binary_logloss: 0.253048	valid_1's auc: 1	valid_1's binary_logloss: 0.255916
[5]	training's auc: 1	training's binary_logloss: 0.204297	valid_1's auc: 1	valid_1's binary_logloss: 0.20524
[6]	training's auc: 1	training's binary_logloss: 0.166066	valid_1's auc: 1	valid_1's binary_logloss: 0.168547
[7]	training's auc: 1	training's binary_logloss: 0.136038	valid_1's auc: 1	valid_1's binary_logloss: 0.13931
[8]	training's auc: 1	training's binary_logloss: 0.111936	valid_1's auc: 1	valid_1's binary_logloss: 0.11498
[9]	training's auc: 1	training's binary_logloss: 0.0921099	valid_1's auc: 1	valid_1's binary_logloss: 0.0993008
[10]	training's auc: 1	training's binary_logloss: 0.0761595	valid_1's auc: 1	valid_1's binary_logloss: 0.0838729
[11]	training's auc: 1	training's binary_logloss: 0.0630697	valid_1's auc: 1	valid_1's binary_logloss: 0.0749432
[12]	training's auc: 1	training's binary_logloss: 0.0524085	valid_1's auc: 1	valid_1's binary_logloss: 0.0637523
[13]	training's auc: 1	training's binary_logloss: 0.0437629	valid_1's auc: 1	valid_1's binary_logloss: 0.0547664
Early stopping, best iteration is:
[3]	training's auc: 1	training's binary_logloss: 0.316467	valid_1's auc: 1	valid_1's binary_logloss: 0.316695
Evaluated only: auc
```

In addition, there is an error in your params. It should be
```
'metric': ['auc','binary'],
```
